### PR TITLE
[14.0][MIG] purchase_discount: Migration to 14.0

### PR DESCRIPTION
**Pull request by Vanneri**
Please Review

### CODE REVIEW

**Comment by kittiu** [comment]
@Vanneri is this will a WIP?

By the way, I am seeing the commit history and looks like you are trying to fix the README.rst, which are not necessary. It will be auto generated out of readme fractions when merged.

Please see this guideline https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0#technical-method-to-migrate-a-module-from-130-to-140-branch

Please use pre-commit too.

Let me know once it is ready for further review. Thank you!

**Comment by Vanneri** [comment]
> @Vanneri is this will a WIP?
> 
> By the way, I am seeing the commit history and looks like you are trying to fix the README.rst, which are not necessary. It will be auto generated out of readme fractions when merged.
> 
> Please see this guideline https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0#technical-method-to-migrate-a-module-from-130-to-140-branch
> 
> Please use pre-commit too.
> 
> Let me know once it is ready for further review. Thank you!

sure let me check

**Comment by Vanneri** [comment]
@kittiu Please review

**Comment by Vanneri** [comment]
@kittiu  @HviorForgeFlow

File: purchase_discount/tests/test_product_supplierinfo_discount.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            self.po_line_1.discount,
+            10,
+            "Incorrect discount for product 6 with partner 3 and qty 1: "
+            "Should be 10%",
+        )
+
+    def test_002_purchase_order_partner_3_qty_10(self):
+        self.po_line_1.write({"product_qty": 10})
+        self.po_line_1._onchange_quantity()
+        self.assertEquals(
```
Comment (kittiu) [inline purchase_discount/tests/test_product_supplierinfo_discount.py]:
```suggestion
        self.assertEqual(
```
Warning from test.

**Comment by kittiu** [comment]
Functional test seem OK.

But please, can you help make just 2 commit according to guideline. https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0#technical-method-to-migrate-a-module-from-130-to-140-branch

It will help us see changes easily. Thanks!

**Comment by Vanneri** [comment]
> Functional test seem OK.
> 
> But please, can you help make just 2 commit according to guideline. https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0#technical-method-to-migrate-a-module-from-130-to-140-branch
> 
> It will help us see changes easily. Thanks!

**Comment by Vanneri** [comment]
> Functional test seem OK.
> 
> But please, can you help make just 2 commit according to guideline. https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0#technical-method-to-migrate-a-module-from-130-to-140-branch
> 
> It will help us see changes easily. Thanks!

Yup done it.

**Review by kittiu — APPROVED** [review]
LGTM

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
Functional review:
default_supplierinfo_discount is not propagated to Purchase Orders properly

**Comment by Vanneri** [comment]
> Functional review:
> default_supplierinfo_discount is not propagated to Purchase Orders properly

can you suggest further changes ?, I didn't tried to change the existing functionality !

**Review by HviorForgeFlow — APPROVED** [review]
default_supplierinfo_discount works fine. I misunderstood the Readme

**Pull request merged**